### PR TITLE
Make the robot keep track of the full last drink so the display can too

### PR DESCRIPTION
**Pull request by audiodude**
Includes some reformatting of the display HTML to make everything fit. Also
changes the format of drink-queue.json.

### CODE REVIEW

File: poll_appengine.py [hunk 1]
```diff
@@ -64,6 +65,9 @@ def __init__(self, base_url, poll_frequency_secs, controller):
     super(self.__class__, self).__init__()
     self.queuefile = 'remote_queue.txt'
     self.jsonqueuefile = 'static/order-queue.json'
+    self.queue_for_display = {}
+    self.finished_drink = None
```
Comment (funrollloops) [inline poll_appengine.py]: Default to an empty dictionary here

File: poll_appengine.py [hunk 2]
```diff
@@ -106,26 +110,25 @@ def post(self, url, **kwargs):
...
-    last_drink_id = None
     while True:
       try:
         queue = json.loads(self.get('next_drink'))
         if not self.controller and queue:
           json_recipe = queue[0]
           drink_id = json_recipe['id']
-          if drink_id == last_drink_id:
-            print "Refusing to remake order %s" % last_drink_id
+          if self.finished_drink and drink_id == self.finished_drink['id']:
```
Comment (funrollloops) [inline poll_appengine.py]: Use finished_drink.get here, the possible race here makes me nervous

File: poll_appengine.py [hunk 3]
```diff
@@ -136,20 +139,28 @@ def run(self):
...
       time.sleep(self.poll_frequency_secs)
 
+  def prepare_queue_for_display(self, queue):
+    self.queue_for_display = {
+      'finished_drink': self.finished_drink,
+      'current_drink': self.current_drink,
+      'queue': queue or []
+    }
+
   def write(self, queue):
```
Comment (funrollloops) [inline poll_appengine.py]: It's kind of wonky that the queue argument is ignored here -- maybe make queue_for_display a local variable in this function, and inline prepare_queue_for_display here?

File: static/display.html [hunk 4]
```diff
@@ -9,56 +9,64 @@
...
-	</div>
-	<div class="expl">
-	  * If you've ordered and don't see your name, talk to a specialist
-	  to get your drink approved.
-	</div>
+	      </div>
+	      <div class="expl">
+	        * If you've ordered and don't see your name, talk to<br>
+          a specialist to get your drink approved.
+	      </div>
```
Comment (funrollloops) [inline static/display.html]: indentation is a little wonky, maybe you're using tabs?

**Review by funrollloops — APPROVED** [review]

File: poll_appengine.py (hunk 1 above)
Comment (audiodude) [inline poll_appengine.py]: Done

File: poll_appengine.py (hunk 2 above)
Comment (audiodude) [inline poll_appengine.py]: Done.

File: static/display.html (hunk 4 above)
Comment (audiodude) [inline static/display.html]: Fixed, thanks!

File: poll_appengine.py (hunk 3 above)
Comment (audiodude) [inline poll_appengine.py]: Got rid of self.queue_for_display as a member variable and inlined it. Good idea!

**Review by audiodude — COMMENTED** [review]
I made additional changes, including fixing bugs that were present when there was no queue, or no current_drink (like clearing out the current drink so the display doesn't display it when the queue is empty).

I also added updating the progress_percent on the syncer's local current_drink copy, so that can be reflected in the display.

http://i.imgur.com/f3XOia3.png

PTAL, thanks.

**Pull request merged**